### PR TITLE
AI Integration: Adds cosmetic fixes

### PR DESCRIPTION
**Pull request by sourabh1007**
## Description

This PR contains below changes:
1. Coverting StatusCode and SubstatusCode as Int from string.
image

2. Renamed _hashed_machine_id_ to _machine_id_
3. Remove "information not available" message and set null instead.

image

7. On GatewayMode create "Internal" type of Activity otherwise "client" type pf Activity. (Verified by @lmolkova )

## Type of change
- [] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryCoreRecorder.cs [hunk 1]
```diff
@@ -92,8 +89,8 @@ public void Record(string key, string value)
                 this.scope.AddAttribute(OpenTelemetryAttributeKeys.NetPeerName, clientContext.Client?.Endpoint?.Host);
 
                 // Client Information
-                this.scope.AddAttribute(OpenTelemetryAttributeKeys.ClientId, clientContext?.Client?.Id ?? OpenTelemetryAttributes.NotAvailable);
-                this.scope.AddAttribute(OpenTelemetryAttributeKeys.UserAgent, clientContext.UserAgent ?? OpenTelemetryAttributes.NotAvailable);
+                this.scope.AddAttribute(OpenTelemetryAttributeKeys.ClientId, clientContext?.Client?.Id);
```
Comment (ealsur) [inline Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryCoreRecorder.cs:92]: Would this then add the attribute with a `null` value now? Should the attribute be there with null or not be there at all?

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryCoreRecorder.cs:92]: I will confirm it after testing it with appinsight but as far as I remember, if null it won't appear in appinsight.

Comment (sourabh1007) [reply Microsoft.Azure.Cosmos/src/Telemetry/OpenTelemetry/OpenTelemetryCoreRecorder.cs:92]: in the above screenshot, you can see I don't have any property which has "null" value e.g. db.cosmosdb.request_content_length_bytes, db.cosmosdb.response_content_length_bytes

**Review by kirankumarkolli — APPROVED** [review]

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**